server/http/test_http/oauth2_complete_test.go

```diff
@@ -271,24 +313,83 @@ func statusOKGraphAPIResponderFunc() {
 	httpmock.RegisterResponder("GET", meRequestURL, meResponder)
 }
 
-func getMockBotPoster(ctrl *gomock.Controller, err error) utils.BotPoster {
+func getBotPosterMessage(displayName string) string {
+	return fmt.Sprintf("### Welcome to the Microsoft Office plugin!\n"+
+		"Here is some info to prove we got you logged in\n"+
+		"Name: %s \n", displayName)
+}
+
+type mockBotPosterConfig struct {
+	userID            string
+	channelID         string
+	message           string
+	postType          string
+	err               error
+	maxTimesDirect    int
+	maxTimesEphemeral int
+}
+
+func getMockBotPoster(ctrl *gomock.Controller, config *mockBotPosterConfig) utils.BotPoster {
 	m := mock_utils.NewMockBotPoster(ctrl)
-	m.EXPECT().PostDirect(gomock.Any(), gomock.Any(), gomock.Any()).Return(err).AnyTimes()
-	m.EXPECT().PostEphemeral(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+	m.EXPECT().PostDirect(config.userID, config.message, config.postType).Return(config.err).MaxTimes(config.maxTimesDirect)
+	m.EXPECT().PostEphemeral(config.userID, config.channelID, config.message).MaxTimes(config.maxTimesEphemeral)
 	return m
 }
 
-func getMockKVStore(ctrl *gomock.Controller, bytes []byte, err error) kvstore.KVStore {
+type mockKVStoreConfig struct {
+	bytes            []byte
+	loadKey          string
+	loadBytes        []byte
+	loadErr          error
+	useStoreAnyKey   bool
+	useStoreAnyValue bool
+	storeKey         string
+	storeValue       []byte
+	storeErr         error
+	deleteKey        string
+	deleteErr        error
+	maxTimesLoad     int
+	maxTimesStore    int
+	maxTimesDelete   int
+}
+
+func getMockKVStore(ctrl *gomock.Controller, config *mockKVStoreConfig) kvstore.KVStore {
 	m := mock_kvstore.NewMockKVStore(ctrl)
-	m.EXPECT().Load(gomock.Any()).Return(bytes, err).AnyTimes()
-	m.EXPECT().Store(gomock.Any(), gomock.Any()).Return(err).AnyTimes()
-	m.EXPECT().Delete(gomock.Any()).Return(err).AnyTimes()
+	m.EXPECT().Load(config.loadKey).Return(config.loadBytes, config.loadErr).MaxTimes(config.maxTimesLoad)
+
+	if config.useStoreAnyValue {
+		if config.useStoreAnyKey {
+			m.EXPECT().Store(gomock.Any(), gomock.Any()).Return(config.storeErr).MaxTimes(config.maxTimesStore)
```

Comment:
Are you sure you want `MaxTimes`, and not just `Times`?

Reply:
Ahh... Yeah we probably want to just use `Times` in this case. I must have missed that in their docs and just went straight to using `MaxTimes`.